refactor: fix type annocations in test/test_data/

We're in the middle of trying to improve the type hints we use in cve-bin-tool.  I previously filed some issues as found with pyright, this particular set of issues is done using mypy.  We're new to type checking, so it's entirely possible that the types currently in the file are incorrect or incomplete. To test the file yourself, you'll need to install mypy (e.g. `pip install mypy`) and then run it with the filename you want to test.   You may need to also run `mypy --install-types` to get it to work on some files.

Note that I'm filing issues for each file individually to help folk who are looking for fun stuff to do as part of hacktoberfest so you can claim one and not interfere with anyone else working on another file.  If you've already got your 4 commits for hacktoberfest, please leave these issues for someone who's still working on theirs!

Here's a list of issues found:

```
test/test_data/unbound.py:4: error: Need type annotation for "mapping_test_data" (hint: "mapping_test_data: List[<type>] = ...")
test/test_data/kerberos_5.py:4: error: Need type annotation for "mapping_test_data" (hint: "mapping_test_data: Dict[<type>, <type>] = ...")
test/test_data/kerberos.py:24: error: Need type annotation for "package_test_data" (hint: "package_test_data: Dict[<type>, <type>] = ...")
```
You can look at the other files in the `test/test_data/` directory to figure out what the `type` should be.  You might need to create a type if we don't have one already set up.


Helpful resources:
* [Python typing documentation](https://docs.python.org/3/library/typing.html)
* [mypy cheat sheet](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)
* [cve-bin-tool's contributor docs](https://github.com/intel/cve-bin-tool/blob/main/CONTRIBUTING.md) 
* If you've contributed to open source but not this project, you might just want our [checklist for a great pull request](https://github.com/intel/cve-bin-tool/blob/main/CONTRIBUTING.md#checklist-for-a-great-pull-request)

Tiny tips for new contributors:
* cve-bin-tool uses https://www.conventionalcommits.org/ style for commit messages, and we have a test that checks the tiltle of your PR.  I've put a good title in the title of this issue (e.g. a good title for this one would be "refactor: resolve mypy errors in test/test_data/") 
* You can make an issue auto close by including a comment "fixes #ISSUENUMBER" in your PR coments where ISSUENUMBER is the actual number of the issue.

--- a/cve_bin_tool/extractor.py
+++ b/cve_bin_tool/extractor.py
@@ -49,6 +49,7 @@ class BaseExtractor:
         # Sets up logger and if we should extract files or just report
         self.logger = logger or LOGGER.getChild(self.__class__.__name__)
         self.error_mode = error_mode
+        self.tempdir = None
         self.file_extractors = {
             self.extract_file_tar: {".tgz", ".tar.gz", ".tar", ".tar.xz", ".tar.bz2"},
             self.extract_file_rpm: {".rpm"},
@@ -309,7 +310,6 @@ class TempDirExtractorContext(BaseExtractor):
 
     def __init__(self, raise_failure=False, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.tempdir = None
         self.raise_failure = raise_failure
 
     async def aio_extract(self, filename):

--- a/test/test_data/grub2.py
+++ b/test/test_data/grub2.py
@@ -1,7 +1,9 @@
 # Copyright (C) 2022 Orange
 # SPDX-License-Identifier: GPL-3.0-or-later
 
-mapping_test_data = []
+from typing import Dict, List
+
+mapping_test_data: List[Dict] = []
 package_test_data = [
     {
         "url": "http://rpmfind.net/linux/opensuse/distribution/leap/15.5/repo/oss/aarch64/",

--- a/test/test_data/kerberos.py
+++ b/test/test_data/kerberos.py
@@ -1,6 +1,8 @@
 # Copyright (C) 2021 Intel Corporation
 # SPDX-License-Identifier: GPL-3.0-or-later
 
+from typing import Dict, List
+
 mapping_test_data = [
     {
         "product": "kerberos",
@@ -21,4 +23,4 @@ mapping_test_data = [
         ],
     },
 ]
-package_test_data = {}
+package_test_data: List[Dict] = []

--- a/test/test_data/kerberos_5.py
+++ b/test/test_data/kerberos_5.py
@@ -1,7 +1,9 @@
 # Copyright (C) 2021 Intel Corporation
 # SPDX-License-Identifier: GPL-3.0-or-later
 
-mapping_test_data = {}
+from typing import Dict, List
+
+mapping_test_data: List[Dict] = []
 package_test_data = [
     {
         "url": "http://mirror.centos.org/centos/7/os/x86_64/Packages/",

--- a/test/test_data/unbound.py
+++ b/test/test_data/unbound.py
@@ -1,7 +1,9 @@
 # Copyright (C) 2022 Orange
 # SPDX-License-Identifier: GPL-3.0-or-later
 
-mapping_test_data = []
+from typing import Dict, List
+
+mapping_test_data: List[Dict] = []
 package_test_data = [
     {
         "url": "http://rpmfind.net/linux/fedora/linux/development/rawhide/Everything/aarch64/os/Packages/u/",

